dockerscripts/docker-entrypoint.sh

```diff
@@ -37,7 +37,63 @@ docker_secrets_env() {
     fi
 }
 
+## Create UID/GID based on available environment variables.
+docker_set_uid_gid() {
+    if  [ -z "$MINIO_GID" ] ||  [ -z "$MINIO_UID" ] ; then
+	    addgroup -g 190 -S minio && adduser -u 190 -S -G minio minio
+    else
+        addgroup -g "$MINIO_GID" -S minio && adduser -u "$MINIO_UID" -S -G minio minio
+    fi
+}
+
+# switch to non root (minio) user if there isn't an existing .minio.sys directory or
+# if its an empty data directory. We dont switch to minio user if there is an existing
+# data directory, this is because we'll need to chown all the files -- which may take
+# too long.
+docker_switch_non_root() {
+    for var in "$@"
+    do
+        if [ "${var}" = "minio" ] || [ "${var}" = "server" ] ; then
+            continue
+        fi
+
+        if [ "${var}" = "gateway" ]; then
+            owner="minio"
+            break
```

Comment:
While we discussed earlier, this was missed. If the arg is `gateway nas`, need to fall through and follow the same flow as server.